simde/arm/neon/reinterpret.h

```diff
@@ -0,0 +1,335 @@
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(f64, f32, float64x1, Element_Target_Return_Type_Short)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(s8,  f32, int8x16, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(s16, f32, int16x8, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(s32, f32, int32x4, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(s64, f32, int64x2, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(u8,  f32, uint8x16, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(u16, f32, uint16x8, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(u32, f32, uint32x4, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A32V7_REINTERPRETQ_GEN(u64, f32, uint64x2, Element_Target_Return_Type_Long)\
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(f64, f32, float64x2, Element_Target_Return_Type_Long)
+
+#define SIMDE_ARM_NEON_REINTERPRET_GEN_F64(Element_Target_Return_Type_Short, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(s8, f64, int8x8, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(s16, f64, int16x4, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(s32, f64, int32x2, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(s64, f64, int64x1, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(u8, f64, uint8x8, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(u16, f64, uint16x4, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(u32, f64, uint32x2, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(u64, f64, uint64x1, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRET_GEN(f32, f64, float32x2, Element_Target_Return_Type_Short) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(s8, f64, int8x16, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(s16, f64, int16x8, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(s32, f64, int32x4, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(s64, f64, int64x2, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(u8, f64, uint8x16, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(u16, f64, uint16x8, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(u32, f64, uint32x4, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(u64, f64, uint64x2, Element_Target_Return_Type_Long) \
+    SIMDE_ARM_NEON_A64V8_REINTERPRETQ_GEN(f32, f64, float32x4, Element_Target_Return_Type_Long)
+
+// These are spaced apart so I can more easily look at them with
+//  `gcc -E`, feel free the change the style.
```

Comment:
FWIW, I usually find it easiest to just pipe the output of `gcc -E` straight to `clang-format`.  Macros like this tend to get squashed into a single line of output, `clang-format` fixes that issue.